internal/xds/rbac/rbac_engine_test.go

```diff
@@ -44,110 +55,189 @@ type addr struct {
+							},
 						},
-						Principals: []*v3rbacpb.Principal{
-							{Identifier: &v3rbacpb.Principal_Any{Any: true}},
+					},
+				},
+				{
+					Action: v3rbacpb.RBAC_DENY,
+					Policies: map[string]*v3rbacpb.Policy{
+						"anyone": {
+							Permissions: []*v3rbacpb.Permission{
+								{Rule: &v3rbacpb.Permission_Any{Any: true}},
+							},
+							Principals: []*v3rbacpb.Principal{
+								{Identifier: &v3rbacpb.Principal_Any{Any: true}},
+							},
 						},
 					},
 				},
 			},
 		},
 		{
-			name: "TestSuccessCaseEnvoyExample",
-			rbacConfig: &v3rbacpb.RBAC{
-				Policies: map[string]*v3rbacpb.Policy{
-					"service-admin": {
-						Permissions: []*v3rbacpb.Permission{
-							{Rule: &v3rbacpb.Permission_Any{Any: true}},
-						},
-						Principals: []*v3rbacpb.Principal{
-							{Identifier: &v3rbacpb.Principal_Authenticated_{Authenticated: &v3rbacpb.Principal_Authenticated{PrincipalName: &v3matcherpb.StringMatcher{MatchPattern: &v3matcherpb.StringMatcher_Exact{Exact: "cluster.local/ns/default/sa/admin"}}}}},
-							{Identifier: &v3rbacpb.Principal_Authenticated_{Authenticated: &v3rbacpb.Principal_Authenticated{PrincipalName: &v3matcherpb.StringMatcher{MatchPattern: &v3matcherpb.StringMatcher_Exact{Exact: "cluster.local/ns/default/sa/superuser"}}}}},
-						},
-					},
-					"product-viewer": {
-						Permissions: []*v3rbacpb.Permission{
-							{Rule: &v3rbacpb.Permission_AndRules{AndRules: &v3rbacpb.Permission_Set{
-								Rules: []*v3rbacpb.Permission{
-									{Rule: &v3rbacpb.Permission_Header{Header: &v3routepb.HeaderMatcher{Name: ":method", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_ExactMatch{ExactMatch: "GET"}}}},
-									{Rule: &v3rbacpb.Permission_UrlPath{UrlPath: &v3matcherpb.PathMatcher{Rule: &v3matcherpb.PathMatcher_Path{Path: &v3matcherpb.StringMatcher{MatchPattern: &v3matcherpb.StringMatcher_Prefix{Prefix: "/products"}}}}}},
-									{Rule: &v3rbacpb.Permission_OrRules{OrRules: &v3rbacpb.Permission_Set{
-										Rules: []*v3rbacpb.Permission{
-											{Rule: &v3rbacpb.Permission_DestinationPort{DestinationPort: 80}},
-											{Rule: &v3rbacpb.Permission_DestinationPort{DestinationPort: 443}},
-										},
-									}}},
-								},
+			name: "TestSuccessCaseSimplePolicySingular",
+			rbacConfigs: []*v3rbacpb.RBAC{
+				{
+					Action: v3rbacpb.RBAC_ALLOW,
+					Policies: map[string]*v3rbacpb.Policy{
+						"localhost-fan": {
+							Permissions: []*v3rbacpb.Permission{
+								{Rule: &v3rbacpb.Permission_DestinationPort{DestinationPort: 8080}},
+								{Rule: &v3rbacpb.Permission_UrlPath{UrlPath: &v3matcherpb.PathMatcher{Rule: &v3matcherpb.PathMatcher_Path{Path: &v3matcherpb.StringMatcher{MatchPattern: &v3matcherpb.StringMatcher_Exact{Exact: "localhost-fan-page"}}}}}},
+							},
+							Principals: []*v3rbacpb.Principal{
+								{Identifier: &v3rbacpb.Principal_Any{Any: true}},
 							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "TestSuccessCaseSimplePolicyMultiple",
```

Comment:
There are a lot of test cases here. But it is not easy to figure out what each test case is specifically targeted towards. 

For example, what does this test case explicitly test? What does a failure in this test case tell you in terms of where the bug could be?

Same about test cases like `TestPrincipalProductViewer` or `TestCertainHeaders` (what does certain mean here).

It's very easy to write a ton of test cases without clearly thinking about what each tests case is for. If we don't do that, it then becomes a maintenance burden. Also, it doesn't serve the second purpose of tests which is that it offers a way to understand the code under test.